core/src/main/java/discord4j/core/object/entity/Webhook.java

```diff
@@ -223,6 +265,75 @@ public String getToken() {
                 .map(data -> new Webhook(gateway, data));
     }
 
+    /**
+     * Requests to edit this webhook.
+     * Does not require the MANAGE_WEBHOOKS permission.
+     *
+     * @param spec A {@link Consumer} that provides a "blank" {@link WebhookEditSpec} to be operated on.
+     * @return A {@link Mono} where, upon successful completion, emits the edited {@link Guild}. If an error is
+     * received, it is emitted through the {@code Mono}.
+     */
+    public Mono<Webhook> editWithToken(final Consumer<? super WebhookEditWithTokenSpec> spec) {
+        return Mono.defer(
+                () -> {
+                    if (!getToken().isPresent()) {
+                        throw new IllegalStateException("Can't edit webhook.");
+                    }
+                    WebhookEditWithTokenSpec mutatedSpec = new WebhookEditWithTokenSpec();
+                    spec.accept(mutatedSpec);
+                    return gateway.getRestClient().getWebhookService()
+                            .modifyWebhookWithToken(getId().asLong(), getToken().get(), mutatedSpec.asRequest());
+                })
+                .map(data -> new Webhook(gateway, data));
+    }
+
+    /**
+     * Executes this webhook without waiting for a confirmation that a message was created.
+     *
+     * @param spec A {@link Consumer} that provides a "blank" {@link WebhookExecuteSpec} to be operated on.
+     * @return A {@link Mono} where, upon successful webhook execution, completes. If the message fails to save,
+     * an error IS NOT emitted through the {@code Mono}.
+     */
+    public Mono<Void> execute(final Consumer<? super WebhookExecuteSpec> spec) {
+        return execute(false, spec).cast(Void.class);
```

Comment:
Uhh, wouldn't that result in ClassCastException? The correct way to convert to `Mono<Void> is `.then()`

Reply:
`Mono.empty().cast(Void.class)` is fine. `then()` might be better semantically, but this won't throw an exception.

Reply:
Ah, didn't know it was supposed to be always empty here.